hoomd/hpmc/update.py

```diff
@@ -512,6 +513,207 @@ def N(self):  # noqa: N802 - allow N as a function name
+        # we attach the shape move in _attach.
+        if new_move is self.shape_move:
+            return
+
+        old_move = self.shape_move
+
+        if new_move is not None and new_move._added:
+            raise ValueError(
+                "Cannot add ShapeMove object to multiple integrators.")
+
+        if old_move is not None:
+            if self._attached:
+                old_move._detach()
+            if self._added:
+                old_move._remove()
+
+        if new_move is None:
+            self._param_dict["shape_move"] = None
+            return
+
+        if self._added:
+            new_move._add(self._simulation)
+        if self._attached:
+            new_move._attach()
+        self._param_dict["shape_move"] = new_move
+
+    def _attach(self):
+        integrator = self._simulation.operations.integrator
+        if not isinstance(integrator, integrate.HPMCIntegrator):
+            raise RuntimeError("The integrator must be a HPMC integrator.")
+        if not integrator._attached:
+            raise RuntimeError("Integrator is not attached yet.")
+
+        updater_cls = None
+        supported_shapes = {
+            'Sphere', 'ConvexPolygon', 'SimplePolygon', 'ConvexPolyhedron',
+            'ConvexSpheropolyhedron', 'Ellipsoid', 'ConvexSpheropolygon',
+            'Polyhedron', 'Sphinx', 'SphereUnion'
+        }
+        integrator_name = integrator.__class__.__name__
+        if integrator_name in supported_shapes:
+            updater_cls = getattr(_hpmc, 'UpdaterShape' + integrator_name)
+        else:
+            raise RuntimeError("Integrator not supported")
+        # TODO: Make this possible
+        # Currently computing the moments of inertia for spheropolyhedra is not implemented
+        # In order to prevent improper usage, we throw an error here. The use of this
+        # updater with spheropolyhedra is currently enabled to allow the use of spherical
+        # depletants
+        if isinstance(integrator, integrate.ConvexSpheropolyhedron):
+            for shape in integrator.shape.values():
+                if shape['sweep_radius'] != 0 and len(shape['vertices']) > 1:
+                    raise RuntimeError(
+                        "Currently alchemical moves with ConvexSpheropolyhedron\
+                    are only enabled for polyhedral and spherical particles.")
+        self._attach_shape_move()
+        self._cpp_obj = updater_cls(self._simulation.state._cpp_sys_def,
+                                    self.trigger, integrator._cpp_obj,
+                                    self.shape_move._cpp_obj)
+        super()._attach()
+
+    @log(category='sequence', requires_run=True)
+    def shape_moves(self):
+        """tuple[int, int]: Count of the accepted and rejected shape moves.
+
+        (0, 0) before the first call to `Simulation.run`.
+        """
+        if self._attached:
+            return self._cpp_obj.getShapeMovesCount()
+        else:
+            return (0, 0)
```

Comment:
```suggestion
        return self._cpp_obj.getShapeMovesCount()
```
The requires_run keyword argument raises an `DataAccessError` when querying before attaching, so no need to have any logic.